src/libs/geometry/src/geometry.cpp

```diff
@@ -229,7 +229,7 @@ void *GEOM_SERVICE_R::malloc(long bytes)
 
 void GEOM_SERVICE_R::free(void *ptr)
 {
-    delete ptr;
+    delete[] static_cast<char*>(ptr);
```

Comment:
What purpose of casting before deletion?

Reply:
this is just not allowed by standard: https://eel.is/c++draft/expr.delete#footnote-69
practically, usually there is no difference, nevertheless it is still UB and treated as a misusage by most modern compilers and SCA tools